From c198158210cd97c1c462af3b505017ff270123f6 Mon Sep 17 00:00:00 2001
From: oliverwen <oliverwen@google.com>
Date: Thu, 14 Jul 2016 09:33:55 -0700
Subject: [PATCH] CHROMIUM: ath10k: Fix spurious wmi train commands after SA
 training completed

FW is not aware of when a training should start and when it should end.
It will send the train packets per host's request. Currently, after the
first training, host is not conveying the training completion information
to FW. As a result, FW would keep sending us the train packet and inturn
we are generating a spurious wmi command configuring train information
with large num_pkts.

Fix this, by sending the wmi command configuring train information with
num_pkts to 0. Now the FW knows that it need not send any more train
packets.

TEST=enable the wmi debug logs, SA logs (0x4002) and set the
smart_debug_level appropriately. You shouldn't see "Train pkt : True"
prints after training is completed. Also you shouldn't see
"wmi peer set smart ant train info" messages after the training is
completed

BUG=chrome-os-partner:51964
CQ-DEPEND=CL:341562

Signed-off-by: oliverwen <oliverwen@google.com>
(cherry picked from commit 82abd692b65e86bd8aec3957103bbab2a9aa3f9c)
(source: wireless-3.18)

Change-Id: I30f9fa88d73f0471dc091ed1bb85643c7a3a6468
Reviewed-on: https://chromium-review.googlesource.com/341563
Commit-Ready: Oliver Wen <oliverwen@chromium.org>
Tested-by: Oliver Wen <oliverwen@chromium.org>
Reviewed-by: Yixiang Li <yixiang@google.com>
---
 drivers/net/wireless/ath/ath10k/smart_ant.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath10k/smart_ant.c b/drivers/net/wireless/ath/ath10k/smart_ant.c
index 04bbf4b7757..886e79084a3 100644
--- a/drivers/net/wireless/ath/ath10k/smart_ant.c
+++ b/drivers/net/wireless/ath/ath10k/smart_ant.c
@@ -447,7 +447,8 @@ ath10k_sant_fill_train_rate_ant(struct ath10k_smart_ant_sta *sa_sta,
 		train_info->antennas[i] = tdata->antenna;
 	}
 
-	train_info->num_pkts = tdata->num_pkts | ATH10K_SMART_ANT_NUM_PKT_MASK;
+	train_info->num_pkts =
+			    tdata->num_pkts ? ATH10K_SMART_ANT_NUM_PKT_MASK : 0;
 }
 
 static void ath10k_sant_reset_train_data(struct ath10k_smart_ant_sta *sa_sta)
-- 
2.43.0

